Feature/IOPS-1551-Profiling England Immunization

KazeemHamzat opened the pull request:
Profiling England Immunization and related assets.

KazeemHamzat committed:
Profiling-England-Immunization

Create-England-Immunization

KazeemHamzat committed:
Fixing Bad Example names

Fixing Bad Example names

APageNHS commented on StructureDefinition/England-Immunization.xml:
```diff
@@ -0,0 +1,347 @@
+    </element>
+    <element id="Immunization.reasonCode">
+      <path value="Immunization.reasonCode" />
+      <definition value="A SNOMED Concept Id. For COVID19 and Flu see https://digital.nhs.uk/developer/api-catalogue/vaccination" />
```
We should always refer to SNOMED as SNOMED CT.

APageNHS commented on valusets/ValueSet-England-ImmunizationExplanationReason.xml:
```diff
@@ -0,0 +1,43 @@
+                <code value="429060002" />
+            </concept>
+            <concept>
+                <code value="281657000" />
```
This is a retired concept - do we want to include it?

KazeemHamzat replied:
Replaced with 171072005

APageNHS commented on valusets/ValueSet-England-VaccinationProcedure.xml:
```diff
@@ -0,0 +1,1907 @@
+            <filter>
+               <property value="constraint"/>
+               <op value="="/>
+               <value value="&lt;&lt;1240681000000103 OR &lt;&lt;90351000119108"/>
```
First concept is retired

KazeemHamzat replied:
Removed 1240681000000103.

APageNHS commented on StructureDefinition/England-Immunization.xml:
```diff
@@ -0,0 +1,347 @@
+      <path value="Immunization.performer.actor" />
+      <type>
+        <code value="Reference" />
+        <profile value="http://hl7.org/fhir/StructureDefinition/Reference" />
```
As per discussion, we shouldn't be adding references

APageNHS commented on valusets/ValueSet-England-VaccinationProcedure.xml:
Have we done a comparison of the result of this expansion, compared to the UK Core  version?

APageNHS commented on valusets/ValueSet-England-VaccinationProcedure.xml:
```diff
@@ -0,0 +1,1907 @@
+    <title value="Interim version of UK Core Vaccination Procedure."/>
+    <status value="draft" />
+    <date value="2023-06-01" />
+    <publisher value="NHS Endland" />
```
Spelling mistake

APageNHS commented on valusets/ValueSet-England-VaccinationProcedure.xml:
```diff
@@ -0,0 +1,1907 @@
+    <url value="https://fhir.nhs.uk/ValueSet/England-VaccinationProcedure" />
+    <version value="1.0.0" />
+    <name value="EnglandVaccinationProcedure" />
+    <title value="Interim version of UK Core Vaccination Procedure."/>
```
Should this have Interim version?

KazeemHamzat replied:
Content is almost same with UKCoreVaccinationProcedure. Thus, we might change the mapping to UKCoreVaccinationProcedure

APageNHS commented on valusets/ValueSets-England-ReasonImmunizationNotAdministered.xml:
Only real difference to UK Core is the addition of the specific SARS-Cov-2 dose declined, and the removal of 591000119102 | Vaccine declined by patient, which includes the SARS-CoV-2 one...

KazeemHamzat replied:
Did text compare, England-ReasonImmunizationNotAdministered is a pure subset of ValueSet 'UKCoreReasonImmunizationNotAdministered. 
1. Now mapped Immunization.statusReason to ValueSet 'UKCoreReasonImmunizationNotAdministered
2. Dropped England-ReasonImmunizationNotAdministered

APageNHS commented on StructureDefinition/England-Immunization.xml:
```diff
@@ -0,0 +1,347 @@
+      <min value="0" />
+      <mustSupport value="true" />
+      <binding>
+        <strength value="required" />
```
Is there a reason that this is required? UK Core standard is for SNOMED based valuesets to be preferred

KazeemHamzat replied:
Now changed to Preferred.

APageNHS commented on StructureDefinition/England-Immunization.xml:
```diff
@@ -0,0 +1,347 @@
+    <element id="Immunization.statusReason">
+      <path value="Immunization.statusReason" />
+      <binding>
+        <strength value="required" />
```
As per above question, is there a reason for required?

APageNHS commented on StructureDefinition/England-Immunization.xml:
```diff
@@ -0,0 +1,347 @@
+      <path value="Immunization.vaccineCode" />
+      <definition value="Where status is `completed` this is Mandatory. This indicates vaccine product administered. This should be a SNOMED CT code from {{link:https://fhir.hl7.org.uk/ValueSet/UKCore-VaccineCode}}&#xD;&#xA;AMPP codes should not be used.&#xD;&#xA;&#xD;&#xA;&lt;table&gt;&#xD;&#xA;&lt;thead&gt;&#xD;&#xA;&lt;th data-no-sort width=&quot;20%&quot;&gt;&#xD;&#xA;Vaccine Reference Set&#x ...
+      <binding>
+        <strength value="extensible" />
```
Is there a reason for extensible? UK Core standard is for SNOMED based to be preferred

KazeemHamzat replied:
Binding Strength changed to preferred.

APageNHS commented on StructureDefinition/England-Immunization.xml:
```diff
@@ -0,0 +1,347 @@
+    </element>
+    <element id="Immunization.route">
+      <path value="Immunization.route" />
+      <definition value="A SNOMED concept ID value from either of the following should be used:&#xD;&#xA;UK “ePrescribing route of administration simple reference set (foundation metadata concept)” (999000051000001100) [UKCore-MedicationDosageRoute](https://simplifier.net/resolve?target=simplifier&amp;scope=uk.nhsdigital.r4&amp;canonical=https://fhir.hl7.org.uk/ValueSet/UKCore-MedicationDosageRou ...
```
SNOMED should be refered to as SNOMED CT

APageNHS commented on StructureDefinition/England-Immunization.xml:
````diff
@@ -0,0 +1,347 @@
+    </element>
+    <element id="Immunization.doseQuantity">
+      <path value="Immunization.doseQuantity" />
+      <definition value="A dm+d (SNOMED) Concept ID value representing the Unit of measure used.&#xD;&#xA;&#xD;&#xA;For COVID-19 a reference set has been published here: https://dd4c.digital.nhs.uk/dd4c/publishedmetadatas/intid/980?size=10&#xD;&#xA;&#xD;&#xA;```json&#xD;&#xA;&#xD;&#xA; &quot;doseQuantity&quot;: {&#xD;&#xA;        &quot;value&quot;: 1,&#xD;&#xA;        &quot;unit&quot;: &quot;pre- ...
````
Should be SNOMED CT, should be Unit of Measure

KazeemHamzat commented on StructureDefinition/England-Immunization.xml:
```diff
@@ -0,0 +1,347 @@
+      <mustSupport value="true" />
+      <binding>
+        <strength value="required" />
+        <description value="(Work in progress) The vaccine preventable disease the dose is being administered for." />
```
@KevinMayfield and @omarisgreat: Can we removed this Work in progress incsription here?

KazeemHamzat committed:
Fixing issues raised on PR-England-Immunization

KazeemHamzat committed:
Reversing Binding Strength to status quo

Returning binding strength for VaccinationProcedure and ReasonImunnizationNotAdministered. Also returning the files back into the project.

digitalhealthmartin commented on examples/England-Immunization-AdministrationofSARSCoV2-SecondDose-Example.xml:
```diff
@@ -0,0 +1,44 @@
+<Immunization xmlns="http://hl7.org/fhir">
+    <id value="England-Immunization-AdministrationofSARSCoV2-SecondDose-Example" />
```
Reflect any filename change

KazeemHamzat commented on examples/England-Bundle-PatientImmunizationQuery-Example.xml:
The file name is ok. The resource type is Bundle not Immunization, and the rule is, file name is prefix after resource type. Entries in a Bundle can be a collection of different resource types. It will be impossible to reflect all the entries in the ID and File name.

KazeemHamzat commented on examples/England-Bundle-PatientImmunizationQuery-Example.xml:
```diff
@@ -0,0 +1,199 @@
+<Bundle xmlns="http://hl7.org/fhir">
+    <id value="England-Bundle-PatientImmunizationQuery-Example" />
```
Same as above

KazeemHamzat commented on examples/England-Bundle-VaccinationQuery-SearchInclude-Example.xml:
```diff
@@ -0,0 +1,218 @@
+<Bundle xmlns="http://hl7.org/fhir">
+    <id value="England-Bundle-VaccinationQuery-SearchInclude-Example" />
```
Same as explained above

KazeemHamzat committed:
Fixing PR Issues raised: England-Immunization

Fixing PR Issues raised: England-Immunization

digitalhealthmartin commented on valusets/ValueSets-England-ReasonImmunizationNotAdministered.xml:
ValueSets -> ValueSet

digitalhealthmartin reviewed (changes requested):
Previous points fixed or adequately addressed. Just found one issue about incorrect filename.

KazeemHamzat committed:
Fixing PR Issues

changed File names prefix toValueSet.

KazeemHamzat committed:
Removing StructureMap (Reference to MedicationRequest)

Removing StructureMap (Reference to MedicationRequest)

KazeemHamzat committed:
Removing files not related England-Immunization

Removing files not related England-Immunization

KazeemHamzat committed:
Removed files not related to England-Immunization

Removed files not related to England-Immunization

digitalhealthmartin reviewed (approved):
Previous issues addressed. Approved

digitalhealthmartin commented on valusets/Valueset-England-OperationOutcomeCodes.xml:
Should this file be here? Actually directly used by Immunization?

digitalhealthmartin reviewed (changes requested):
One file that looks like it might not belong

KazeemHamzat committed:
Delete Valueset-England-OperationOutcomeCodes.xml

Removing unrelated files

digitalhealthmartin reviewed (dismissed):
Fixes made. Approved

KazeemHamzat committed:
Fixing PR: Adding CapabilityStatement

Fixing PR: Adding CapabilityStatement

KazeemHamzat committed:
Adding CS: England-Immunization

Adding CS: England-Immunization

digitalhealthmartin committed:
Merge branch 'DSTU2_develop' into Feature/IOPS-1551-Profiling-England-Immunization
